### PR TITLE
unescape username & password in shh connection

### PR DESCRIPTION
**Pull request by naishx**
added Uri.UnescapeDataString for ssh username and passwort in ssh connection. So it's possible to use domain username.

Example :

```
var usernameWithDomain = Uri.EscapeDataString(@"domain\bob");
var sshConnectionString = "ssh://" + usernameWithDomain + ":password@host";
```

Will create:
"ssh://domain%5Cbob:password@host"

Before this change the methode CreateConnectionInfo() did not unescape username and password, so you could not connect.

### CODE REVIEW

**Comment by niemyjski** [comment]
@naishx thanks for the pr. I had a quick question. Was there any error with just leaving the domain as is `domain\bob`? Was there there an error? if so could you paste It in here.

File: src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs [hunk 1]
```diff
@@ -211,18 +211,22 @@ public SshNetFileStorage(Builder<SshNetFileStorageOptionsBuilder, SshNetFileStor
 
             string[] userParts = uri.UserInfo.Split(new [] { ':' }, StringSplitOptions.RemoveEmptyEntries);
             string username = userParts.First();
```
Comment (niemyjski) [inline src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs]: Can we just unescape the username and password by default `string username = Uri.UnescapeDataString(userParts.First())`

**Comment by niemyjski** [comment]
Awesome thanks for that, I left a single comment about just using a single variable for unescaped. We'll get this merged in and out in the next release (7.2)

**Comment by naishx** [comment]
When can i expect the next release?

**Comment by niemyjski** [comment]
It should be shortly (by next week I'd think), we've been making some changes to the storage interface to make paging lots easier for file lists. In the mean time we have nightly builds that get pushed out to myget.

**Pull request merged**